Minor edits to Airflow glossary (#2587)
astronomer/docs: learn/airflow-glossary.md

---
sidebar_label: 'Airflow Glossary'
title: 'Apache Airflow glossary'
id: airflow-glossary
description: A list of key Apache Airflow terms and definitions that can help you learn important concepts.
---

Whether you're new to Airflow or not, use this glossary to quickly reference key Airflow terms, components, and concepts.

| Term | Definition |
|------|-------------|

> | Airflow Connection| An [Airflow connection](connections.md) is a set of configurations and credentials that allows Airflow to connect with other tools in the data ecosystem. For example, you need to create a Snowflake connection to push data into Snowflake. |

Reviewer: ```suggestion
| Airflow Connection| An [Airflow connection](connections.md) is a set of configurations and credentials that allows Airflow to connect with other tools in the data ecosystem. For example, you need to create a Snowflake connection to access data in Snowflake. |
```

Just a small suggestion, feel free to ignore! I just think accessing data is a slightly more general term than loading it. But both work as an example of course :)

> | Airflow Connection| An [Airflow connection](connections.md) is a set of configurations and credentials that allows Airflow to connect with other tools in the data ecosystem. For example, you need to create a Snowflake connection to access data in Snowflake. |

Reviewer 2: ```suggestion
| Airflow Connection| An [Airflow connection](connections.md) is a set of configurations and credentials that allows Airflow to connect with external tools. |
```
Not sure the "For example" statement here offers any more context than the base definition besides throwing out an example of a "tool".

> | Airflow UI| The [Airflow UI](airflow-ui.md) is the primary interface for managing DAG and task runs. You can open and use the Airflow UI in your web browser. It contains pages for modifying, monitoring, and troubleshooting an Airflow environment. |

Reviewer 2: ```suggestion
| Airflow UI| The [Airflow UI](airflow-ui.md) is the primary visual interface for managing DAG and task runs. It contains pages for modifying, monitoring, and troubleshooting an Airflow environment. |
```
Feels like the additional content here doesn't fully capture information that we're introducing. Okay, I can use the Airflow UI in the web browser - but how? Is it powered by Airflow itself? How is it hosted? 

It's okay to have some of these details explained in the linked documentation.

| Airflow Variable| An [Airflow variable](airflow-variables.md) is a generic key value pair, such as an API key or file path, that's stored in the Airflow metadata database and that you can reference in a DAG. |
| Apache Airflow | Apache Airflow is an open source tool for programmatically authoring, scheduling, and monitoring data pipelines written in Python. Airflow is scalable, configurable, and the industry standard for managing workflows across your ecosystem. |
| Dataset| A [dataset](airflow-datasets.md) is a logical grouping of data consumed or produced by tasks in an Airflow DAG. It can be a table, a file, a blob, or a dataframe. Datasets can be used to schedule DAGs with dataset-driven scheduling. |
| Decorator| In Python, decorators are functions that take another function as an argument and extend the behavior of that function. In Airflow, [decorators](airflow-decorators.md) provide a simpler way to define Airflow tasks and DAGs compared to traditional operators. |

> | Deferrable operator | A [deferrable operator](deferrable-operators.md), also known as an async operator, is an operator that suspends itself while waiting for its condition to be met and resumes on receiving the job status. Tasks that use deferrable operators consume worker resources more efficiently because they do not occupy a worker slot when they are in a deferred state. Instead, deferred tasks use the triggerer to poll for job statuses. |

Reviewer 2: ```suggestion
| Deferrable operator | A [deferrable operator](deferrable-operators.md), also known as an async operator, is an operator that suspends itself while waiting for its condition to be met and resumes on receiving the job status. Tasks that use deferrable operators consume resources more efficiently than sensors because they do not occupy a worker slot when they are in a deferred state. Instead, deferred tasks use the triggerer to poll for job statuses. |
```

> | Task dependency| A task dependency is an instruction that defines whether a task must be completed either before or after another task in the same DAG. Dependencies are defined in DAG code either explicitly with bitshift operators or implicitly with the TaskFlow API. The term 'dependency' is often also used to describe a Python or OS-level package. |

Reviewer: ```suggestion
| Task dependency| A task dependency is an instruction that defines whether a task must be completed either before or after another task in the same DAG. Dependencies are defined in DAG code either explicitly with bitshift operators and utility functions or implicitly with the TaskFlow API. The term 'dependency' is often also used to describe a Python or OS-level package. |
```

On the fence about my own suggestion here. I know historically dependency functions like `chain()` have not really been highlighted much. But 2.7 will add a new one so I wonder if we should talk about them more. If that distinction is needed in the glossary, I am not sure about :)

> | Task dependency| A task dependency is an instruction that defines whether a task must be completed either before or after another task in the same DAG. Dependencies are defined in DAG code either explicitly with bitshift operators and utility functions or implicitly with the TaskFlow API. The term 'dependency' is often also used to describe a Python or OS-level package. |

Reviewer 2: ```suggestion
| Task dependency| A task dependency is an instruction that defines whether a task must be completed either before or after another task in the same DAG. Task dependencies are defined in DAG code either explicitly with bitshift operators or implicitly with the TaskFlow API.  |
```
Where is "utility functions" coming from? Never heard that term in relation to task dependencies. 

Also I don't think we need to clarify the synonym here. It makes it sound like these two definitions are related, when they're not.

| Docker image| An Airflow [Docker image](https://www.techtarget.com/searchitoperations/definition/Docker-image) is a template used to build the Docker containers which run Airflow components and execute DAG code. Both Apache Airflow and Astronomer distribute Docker images for Airflow with different build instructions and pre-installed packages. |

> | Dynamic DAG | A [Dynamic DAG](dynamically-generating-dags.md) is a DAG run that is generated automatically during the parsing of the DAG file based on some condition. |

Reviewer: ```suggestion
| Dynamic DAG | A [Dynamic DAG](dynamically-generating-dags.md) is a DAG that is generated automatically during the parsing of the `dags` folder based Python logic. You can dynamically create DAGs based on code in a single or multiple Python files or by using tools like `gusty` or `dag-factory`.  |
```

I think Dynamic DAGs still refers to the old static "parsing code to create DAGs" pattern. @kentdanas is still still correct, now I am unsure 😅 
Also adding the dag-factory keyword for SEO maybe? It is still highly used (even if gusty has more features)

> | Dynamic DAG | A [Dynamic DAG](dynamically-generating-dags.md) is a DAG run that is generated automatically during the parsing of the DAG file based on some condition. |

Author: ```suggestion
| Dynamic DAG | A [Dynamic DAG](dynamically-generating-dags.md) is a DAG that is generated automatically when the scheduler parses the `dags` folder. You can dynamically create DAGs based on code in one or more Python files or by using tools like `gusty` or `dag-factory`.  |
```

@TJaniF This better?

Reviewer: @paolaperaza looks great! :)

| Dynamic task | A [Dynamic task](dynamic-tasks.md#dynamic-task-concepts) is a task instance that's generated at runtime based on a set of parameters in DAG code. Dynamic task mapping, the Airflow feature that creates dynamic tasks, allows users to create an arbitrary number of parallel tasks at runtime based on an input parameter. |

> | Environment variable| An environment variable is a key-value pair that can be used to define an Airflow environment configuration. This includes Airflow values defined in the airflow.cfg file or any custom value. |

Reviewer 2: ```suggestion
| Environment variable| An environment variable is a key-value pair that can be used to define an Airflow environment configuration. You typically set environment variables in the `airflow.cfg` file. |
```

> | Executor| An [Executor](airflow-executors-explained.md) is a core process within the Airflow scheduler that is responsible for assigning scheduled tasks to a worker process that will complete the function of a task. Airflow supports multiple executors that you can choose from depending on your use case. |

Reviewer 2: ```suggestion
| Executor| An [executor](airflow-executors-explained.md) is a core process within the Airflow scheduler that is responsible for assigning scheduled tasks to a worker process that will complete the function of a task. Airflow supports multiple executors that differ based on the types of workers they use. |
```

| Hook| A [hook](what-is-a-hook.md) is an abstraction of a specific API that allows Airflow to interact with an external system. Hooks are built into many operators, but they can also be used directly in DAG code. |
| Jinja Template| [Jinja templating](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/operators.html#jinja-templating) is a format that is used to pass dynamic information into task instances at runtime. A jinja templated value is enclosed in double curly braces. |

> | Notifier| A [notifier](error-notifications-in-airflow#notifiers) is a custom class that is pre-built into some provider packages and can be used to send notifications to tools like Slack or PagerDuty when a task reaches a particular state. |

Reviewer 2: ```suggestion
| Notifier| A [notifier](error-notifications-in-airflow#notifiers) is a custom class that is pre-built into some provider packages and can be used to send notifications to tools like Slack or PagerDuty. |
```

| Sensor| An [Airflow Sensor](what-is-a-sensor.md) is a special kind of operator that is designed to wait for something to happen. When sensors run, they check to see if a certain condition is met before they are marked successful and let their downstream tasks execute. |
| Task| A task is the basic unit of execution in Airflow. Tasks are arranged into DAGs, and then have upstream and downstream dependencies set between them to express the order in which they should run. |
| Task group| A [task group](task-groups.md) is a way to visually organize a group of tasks in the Airflow UI. Task groups are defined in DAG code and render as groupings in the **Graph** view of the Airflow UI.  |
| TaskFlow API| The [TaskFlow API](dags.md#writing-dags-with-the-taskflow-api) is a framework for using decorators to define DAGs and tasks. Compared to using traditional operators, using the TaskFlow API simplifies the process for passing data between tasks and defining dependencies. |
| Triggerer| The triggerer is an optional Airflow component responsible for running [deferrable operators](deferrable-operators.md#terms-and-concepts) when they're in a deferred state. |

> | Webserver| The webserver is the core Airflow component that serves the Airflow UI. It is a Flask server running with Gunicorn. |

Reviewer 2: ```suggestion
| Webserver| The webserver is the Airflow component that serves the Airflow UI. It is a Flask server running with Gunicorn. |
```

> | XCom| [XCom](airflow-passing-data-between-tasks.md#xcom) is a core Airflow feature that allows you to exchange task metadata or small amounts of data between tasks. XComs are defined by a key, value, and timestamp. |

Reviewer 2: ```suggestion
| XCom| [XCom](airflow-passing-data-between-tasks.md#xcom) is an Airflow feature that allows you to exchange task metadata or small amounts of data between tasks. XComs are defined by a key, value, and timestamp. |
```
